WordPress/src/main/java/org/wordpress/android/ui/uploads/MediaUploadHandler.java

```diff
@@ -251,14 +262,27 @@ private void cancelUpload(MediaModel oneUpload, boolean delete) {
         }
     }
 
-    private void dispatchUploadAction(@NonNull final MediaModel media, @NonNull final SiteModel site) {
+    private void dispatchUploadAction(@NonNull final MediaModel media, boolean optimize) {
+        SiteModel site = mSiteStore.getSiteByLocalId(media.getLocalSiteId());
+
+        // somehow lost our reference to the site, complete this action
+        if (site == null) {
+            AppLog.w(T.MEDIA, "MediaUploadHandler > Unexpected state, site is null. Skipping this request.");
+            notifyServiceIfUploadsComplete();
+            return;
+        }
+
         AppLog.i(T.MEDIA, "MediaUploadHandler > Dispatching upload action for media with local id: "
                 + media.getId() + " and path: " + media.getFilePath());
         sInProgressUploads.add(media);
```

Comment:
One thing I found while testing this is this: uploading a new video never stops uploading (the notification is not dismissed, and the post in the posts list is shown as uploading forever, can’t be cancelled unless you delete the post altogether)

The problem is the `media` model is added twice to `sInProgressUploads` because it is called again in `onVideoOptimizationCompleted()` in https://github.com/wordpress-mobile/WordPress-Android/pull/6473/files#diff-fd1074f533e624627941f15b88dd3277R403

We could save this situation by making sure we don’t add existing items to the `sInProgressUploads` array, i.e. like this:

```
            AppLog.i(T.MEDIA, "MediaUploadHandler > Dispatching upload action for media with local id: "
                    + media.getId() + " and path: " + media.getFilePath());
	     safeAddMediaToInProgressUploads(media);

```

and

``` 
    private void safeAddMediaToInProgressUploads(MediaModel mediaToAdd) {
        synchronized (sInProgressUploads) {
            for (MediaModel media : sInProgressUploads) {
                if (media.getId() == mediaToAdd.getId()) {
                    return;
                }
            }
            sInProgressUploads.add(mediaToAdd);
        }
    }
```

Or maybe making `sInProgressUploads` a `Map` instead of a `List`, although I prefer to keep it a `List`.